rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -113,9 +113,10 @@ Recorder::get_requested_or_available_topics(const RecordOptions & record_options
   std::regex exclude_regex(record_options.exclude);
   bool take = record_options.all;
   for (const auto & kv : unfiltered_topics) {
-    take = std::regex_search(kv.first, topic_regex);
+    // regex_match returns false for 'empty' regex
+    if (!record_options.regex.empty()) {take = std::regex_match(kv.first, topic_regex);}
```

Comment:
Can you expand this onto a separate line please?

Reply:
Done :+1: